### PR TITLE
Use integer flags to denote species units instead of strings

### PR DESCRIPTION
**Pull request by yantosca**
### Name and Institution (Required)

Name: Bob Yantosca
Institution: Harvard + GCST

### Confirm you have reviewed the following documentation

- [x] [Contributing guidelines](https://geos-chem.readthedocs.io/en/stable/reference/CONTRIBUTING.html)

### Describe the update

This update will improve how unit conversions are done in GEOS-Chem:
- [x] Use integer unit flags instead of strings.  This will improve runtime as integer comparisons are much faster than string comparisons.

### Expected changes
This should be a zero-diff update, as we do not plan to introduce any differences.  However, it may be that this ends up introducing differences at the level of numerical noise.

### Reference(s)

N/A

### Related Github Issue(s)

- #1984

### CODE REVIEW

**Comment by yantosca** [comment]
Integration tests (14.2.2_r1) now running.

**Comment by yantosca** [comment]
All GCHP compilation tests failed.  GEOS-Chem Classic tests are proceeding.
```console
Compiliation tests:
------------------------------------------------------------------------------
GCHP................................................Configure & Build.....FAIL
GCHP w/ carbon gases (as a KPP mechanism)...........Configure & Build.....FAIL
GCHP with RRTMG.....................................Configure & Build.....FAIL
```

**Comment by yantosca** [comment]
All GEOS-Chem Classic integration tests passed (except for TOMAS, which will be fixed soon):
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #ad68c48 GEOS-Chem 14.2.1 and HEMCO 3.7.1 release
GEOS-Chem #eb803ca49 Fix several issues with integer unit parameters in newer code
HEMCO     #00aaf65 Update HEMCO 3.7.1 release date in CHANGELOG.md

Using 24 OpenMP threads
Number of execution tests: 27

Submitted as SLURM job: 4591914
=============================================================================
 
Summary of test results:
------------------------------------------------------------------------------
Execution tests passed: 25
Execution tests failed: 2
Execution tests not yet completed: 0
```

**Comment by yantosca** [comment]
Also all GEOS-Chem Classic integration tests were zero-diff w/r/t the prior integration test (for PR #1977), except for the usual suspects:
```console
Checking gc_4x5_merra2_fullchem_APM
   -> 2 differences found in OutputDir
      * GCC_r30//rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.Metrics.20190701_0000z.nc4 
        GCC_r1/rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.Metrics.20190701_0000z.nc4 
      * GCC_r30//rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.SpeciesConc.20190701_0000z.nc4 
        GCC_r1/rundirs/gc_4x5_merra2_fullchem_APM/OutputDir/GEOSChem.SpeciesConc.20190701_0000z.nc4 
   -> 1 difference found in Restarts
      * GCC_r30//rundirs/gc_4x5_merra2_fullchem_APM/Restarts/GEOSChem.Restart.20190701_0100z.nc4 
        GCC_r1/rundirs/gc_4x5_merra2_fullchem_APM/Restarts/GEOSChem.Restart.20190701_0100z.nc4 


Checking gc_4x5_merra2_fullchem_RRTMG
   -> 1 difference found in OutputDir
      * GCC_r30//rundirs/gc_4x5_merra2_fullchem_RRTMG/OutputDir/GEOSChem.RRTMG.20190701_0000z.nc4 
        GCC_r1/rundirs/gc_4x5_merra2_fullchem_RRTMG/OutputDir/GEOSChem.RRTMG.20190701_0000z.nc4 
   -> No differences in Restarts
```
These simulations are known to have parallelization issues.

File: GeosCore/emissions_mod.F90
Comment (msulprizio) [inline GeosCore/emissions_mod.F90:391]: This routine was deleted in https://github.com/geoschem/geos-chem/pull/1816. It must have snuck back in with a merge. Please remove again.

**Review by msulprizio — CHANGES_REQUESTED** [review]
This PR should be good to merge after removing the obsolete MMR_Compute_Flux routine from emissions_mod.F90.

**Comment by yantosca** [comment]
Good catch @msulprizio.  MMR_Compute_Flux is now removed again in commit .189cc1c0de6f4cfe4941523ea4c15aa3d5c4bc68

**Review by lizziel — APPROVED** [review]
Looks good to me! In the future could you put formatting changes in a different PR? It was hard to hone in on the changes with so many indentation etc changes.

**Review by msulprizio — APPROVED** [review]

**Comment by yantosca** [comment]
> Looks good to me! In the future could you put formatting changes in a different PR? It was hard to hone in on the changes with so many indentation etc changes.

Will do @lizziel.

**Comment by yantosca** [comment]
> Looks good to me! In the future could you put formatting changes in a different PR? It was hard to hone in on the changes with so many indentation etc changes.

Will do @lizziel.

**Comment by yantosca** [comment]
After merging locally, all integration tests passed (except TOMAS) and were zero-diff w/r/t 14.2.2. (except APM and RRTMG diagnostic output).

**Pull request merged**
